refpolicy: add variant that builds modular policy

flyn-org opened the pull request:
This adds a variant of refpolicy that builds the modular version of the SELinux policy. There are a few reasons why this should not be the default case, namely the resource requirements of a modular policy along with the read-only nature of /var on OpenWrt. @doverride noted this elsewhere (https://github.com/openwrt/openwrt/pull/3472#issuecomment-705585219). Despite this, I have found the modular policy useful for experimentation.

Signed-off-by: W. Michael Petullo <mike@flyn.org>

adschm commented:
Please add a commit message describing what you change and why. Although there unfortunately have been precedents without, the commit message should be considered mandatory.

flyn-org committed:
refpolicy: add variant that builds modular policy

This adds a variant of refpolicy that builds the modular form of the
policy. While this requires more memory on the target device, along with
some tricks to deal with OpenWrt's volatile /var directory, it is useful
for experiementing with SELinux policy.

Signed-off-by: W. Michael Petullo <mike@flyn.org>

dangowrt commented:
Cherry-picked onto master. Thanks!
